build.gradle

```diff
@@ -372,13 +375,43 @@ task sourcesJar(type: Jar) {
     classifier = 'sources'
 }
 
-// This is a hack to disable the java 8 default javadoc lint until we fix the html formatting
 if (JavaVersion.current().isJava8Compatible()) {
     tasks.withType(Javadoc) {
-        options.addStringOption('Xdoclint:none', '-quiet')
+        // do this for all javadoc tasks, inlcuding gatkDocs
+        options.addStringOption('Xdoclint:none')
     }
 }
 
+javadoc {
+    // This is a hack to disable the java 8 default javadoc lint until we fix the html formatting
+    // We only want to do this for the javadoc task, not gatkDoc
+    options.addStringOption('Xdoclint:none', '-quiet')
+}
+
+// Generate GATK Online Doc
+task gatkDoc(type: Javadoc, dependsOn: shadowJar) {
+    source = sourceSets.main.allJava
+    classpath = sourceSets.main.runtimeClasspath
+
+    // Since the custom doclet we run is basically another entry point for gatk, it needs all of the gatk
+    // dependencies on the docletpath (i.e., Barclay , etc), so we depend on and use the shadowJar. Also,
+    // add the resources folder to ensure log4j is configured.
+    options.docletpath = [file(shadowJar.archivePath)] + files("src/main/resources")
```

Comment:
I originally tried setting the docletpath the way [Picard](https://github.com/broadinstitute/picard/blob/master/build.gradle#L232) does, which I think is similar to what you're suggesting, and had other problems. compileTime classpath seems to have all of the *dependencies*, but the doclet class itself isn't found that way. However, I just tried reverting to runtimeClassPath and that seems to work, so I'll just need to update this to manually include the tools.jar file.